lib/platform/linux/platform.c

```diff
@@ -323,6 +325,7 @@ bool Platform_init()
 
 #if MAX_DURATION_POLL_FAIL_S != 0
     m_last_successful_poll_ts = get_timestamp_s();
+    max_poll_fail_duration = MAX_DURATION_POLL_FAIL_S;
```

Comment:
Maybe you could rename MAX_DURATION_POLL_FAIL_S by DEFAULT_MAX_DURATION_POLL_FAIL_S
And remove all the parts that are under #if MAX_DURATION_POLL_FAIL_S != 0 and replace it by if (max_poll_fail_duration > 0)

It was implemented as a compile configuration setting but as you now implemented an API, let's just remove the compile time version